Zend/zend_gc.c

```diff
@@ -365,13 +374,38 @@ ZEND_API void ZEND_FASTCALL gc_remove_from_buffer(zend_refcounted *ref)
 	}
 }
 
-static void gc_scan_black(zend_refcounted *ref)
+static gc_refcounted_stack* gc_refcounted_stack_push(gc_refcounted_stack *head_element, zend_refcounted *ref)
+{
+	gc_refcounted_stack *new_element;
+	new_element = (gc_refcounted_stack*) emalloc(sizeof(gc_refcounted_stack));
+
+	new_element->ref = ref;
+	new_element->next = head_element;
+
+	return new_element;
+}
+
+static gc_refcounted_stack* gc_refcounted_stack_pop(gc_refcounted_stack *head_element)
+{
+	gc_refcounted_stack *next_element;
+	next_element = head_element->next;
+
+	efree(head_element);
+
+	return next_element;
+}
+
+static gc_refcounted_stack* gc_scan_black(zend_refcounted *ref, gc_refcounted_stack *stack)
```

Comment:
I think it would be more elegant to accept a `gc_refcounted_stack **` here and avoid changing the return points everywhere.

Reply:
ok, this way it will be in sync with `gc_collect_white`

Reply:
fixed in another way and avoided changing the return code lines.